pkg/testcase/secretsencrypt.go

```diff
@@ -0,0 +1,161 @@
+package testcase
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/rancher/distros-test-framework/shared"
+
+	. "github.com/onsi/gomega"
+)
+
+func TestSecretsEncryption() {
+	nodes, errGetNodes := shared.GetNodesByRoles("etcd", "control-plane")
+	Expect(nodes).NotTo(BeEmpty())
+	Expect(errGetNodes).NotTo(HaveOccurred(), "error getting etcd/control-plane nodes")
+
+	product, err := shared.Product()
+	Expect(err).NotTo(HaveOccurred(), "error getting product from config")
+
+	errSecret := shared.CreateSecret("secret1", "default", false)
+	Expect(errSecret).NotTo(HaveOccurred(), "error creating secret")
+
+	shared.LogLevel("INFO", "TEST: 'CLASSIC' Secrets Encryption method")
+
+	index := len(nodes) - 1
+	cpIp := nodes[index].ExternalIP
+	secretsEncryptOps("prepare", product, cpIp, nodes)
+	secretsEncryptOps("rotate", product, cpIp, nodes)
+	secretsEncryptOps("reencrypt", product, cpIp, nodes)
+
+	if strings.Contains(os.Getenv("TEST_TYPE"), "both") {
+		shared.LogLevel("INFO", "TEST: 'NEW' Secrets Encryption method")
+		secretsEncryptOps("rotate-keys", product, cpIp, nodes)
+	}
+}
+
+func secretsEncryptOps(action, product, cpIp string, nodes []shared.Node) {
+	shared.LogLevel("INFO", fmt.Sprintf("TEST: Secrets-Encryption: %s", action))
+	_, errStatusB4 := shared.SecretEncryptOps("status", cpIp, product)
+	Expect(errStatusB4).NotTo(HaveOccurred(), "error getting secret-encryption status before action")
+
+	stdOutput, err := shared.SecretEncryptOps(action, cpIp, product)
+	Expect(err).NotTo(HaveOccurred(), "error: secret-encryption: "+action)
+	verifyActionStdOut(action, stdOutput)
+	if (action == "reencrypt") || (action == "rotate-keys") {
+		shared.LogLevel("DEBUG", "reencrypt op needs some time to complete - Sleep for 20 seconds before service restarts")
+		time.Sleep(20 * time.Second) // Wait for reencrypt action to complete before restarting services
+	}
+	for i, node := range nodes {
+		nodearr := []string{node.ExternalIP}
+		nodeIp, errRestart := shared.ManageService(product, "restart", "server", nodearr)
+		Expect(errRestart).NotTo(HaveOccurred(), "error restart service for node: "+nodeIp)
+		// Order of reboot matters. Etcd first then control plane nodes.
+		// Little lag needed between node restarts to avoid issues.
+		waitEtcdErr := shared.WaitForPodsRunning(5, 4)
+		if waitEtcdErr != nil {
+			shared.LogLevel("WARN", "pods not up after 20 seconds.")
+			if i != len(nodes)-1 {
```

Comment:
nope it was for logging when the restarts where continuing and it would not have logged for the last time - when there is no more node restarts happening. but i removed this log line to avoid any confusion for anyone.